webserver/views/api/v1/datasets.py

```diff
@@ -217,7 +217,18 @@ def add_recordings(dataset_id):
 
     :resheader Content-Type: *application/json*
     """
-    raise NotImplementedError
+    class_dict = request.get_json()
+    if not class_dict:
+        raise api_exceptions.APIBadRequest("Data must be submitted in JSON format.")
+    try:
+        db.dataset.add_recordings(class_dict, dataset_id)
+    except dataset_validator.ValidationException as e:
+        raise api_exceptions.APIBadRequest(e.message)
```

Comment:
You are catching an exception here, but this exception is not raised in the dataset method.